src/uwtools/api/config.py

```diff
@@ -68,10 +97,35 @@ def translate(
     return False
 
 
-def validate(input_file: DefinitePath, input_format: str, schema_file: DefinitePath) -> bool:
+def validate(
```

Comment:
`validate()`, as extended for API use, can operate on an in-memory `dict` or `Config` objects, as well as the existing support for file/stream IO. The `YAMLConfig` class is used for validating a `dict` object.